config/configgrpc/configgrpc.go

```diff
@@ -366,3 +372,24 @@ func GetGRPCCompressionKey(compressionType string) string {
 	}
 	return CompressionUnsupported
 }
+
+// enhanceWithClientInformation intercepts the incoming RPC, replacing the incoming context with one that includes
+// a client.Info, potentially with the peer's address.
+func enhanceWithClientInformation(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+	return handler(contextWithClient(ctx), req)
+}
+
+func enhanceStreamWithClientInformation(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	// TODO: how to change the context in this case?
```

Comment:
```
// wrappedServerStream is a thin wrapper around grpc.ServerStream that allows modifying context.
type wrappedServerStream struct {
	grpc.ServerStream
	// WrappedContext is the wrapper's own Context. You can assign it.
	WrappedContext context.Context
}

// Context returns the wrapper's WrappedContext, overwriting the nested grpc.ServerStream.Context()
func (w *wrappedServerStream) Context() context.Context {
	return w.WrappedContext
}
```

Reply:
Why didn't I think about this before...

Reply:
Turns out, grpc-go-middleware already has something just like this. Given that there are a few testing facilities from it that I could use, I'll use their wrappers as well for this.

https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/wrappers.go#L25